File: src/main/Java/boj_step6/boj1316.java
import java.io.*;
import java.util.LinkedList;

public class boj1316 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        int count = 0;
        while (n-- > 0) {
            int tmp = 0;
            LinkedList<Character> list = new LinkedList<>();
            char[] line = reader.readLine().toCharArray();
            list.add(line[0]);
            for (int i = 1; i < line.length; i++) {
                if (line[i] != line[i - 1]) {
                    if (list.contains(line[i])) {
                        tmp++;
                        break;
                    } else list.add(line[i]);
                }
            }
            if (tmp == 0) count++;
        }
        writer.append(String.valueOf(count));
        writer.close();
    }
}

File: src/main/kotlin/boj1316.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    var count = 0
    repeat(r.readLine().toInt()) {
        var tmp = 0
        val list = mutableListOf<Char>()
        val str = r.readLine()
        list.add(str.get(0))
        for (i in 1 until str.length) {
            if (str.get(i) == str.get(i - 1)) continue
            else {
                if (str.get(i) in list) {
                    tmp++
                    break
                } else list.add(str.get(i))
            }
        }
        if (tmp == 0) count++
    }
    w.write(count.toString())
    w.close()
}